### PR TITLE
update machine learning compute portion of the machine learning services swagger

### PR DESCRIPTION
**Pull request by dagald**
Update Machine Learning Compute portion of the swagger.
Examples updated

Run OAV and RestAPI validation.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json [hunk 1]
```diff
@@ -562,7 +555,6 @@
           "MachineLearningComputes"
         ],
         "operationId": "MachineLearningCompute_Delete",
-        "x-ms-long-running-operation": true,
```
Comment (dsgouda) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: Why is this not long running? 201 status corresponds to `Created` which is incorrect for a `DELETE` operation

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json [hunk 2]
```diff
@@ -608,15 +597,64 @@
...
+          {
+            "$ref": "#/parameters/ComputeNameParameter"
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Compute System Update initiated.",
```
Comment (dsgouda) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: Is this and update for the `MachineLearningCompute` resource?
`Update` is usually a `PATCH` operation. Did this service undergo an API review?

Comment (dsgouda) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: Also, 202 corresponds to `Accepted` which ideally should be a long running operation

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/examples/createBasicAKSCompute.json [hunk 3]
```diff
@@ -13,18 +13,6 @@
         }
     },
     "responses": {
-        "200": {
```
Comment (salilbapat) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/examples/createBasicAKSCompute.json]: Why is this removed? Same questions for other such removals of 200 responses

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left some comments

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json [hunk 4]
```diff
@@ -500,7 +500,6 @@
           "MachineLearningComputes"
         ],
         "operationId": "MachineLearningCompute_CreateOrUpdate",
-        "x-ms-long-running-operation": true,
```
Comment (salilbapat) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: Is this header not used anymore?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/examples/createBasicAKSCompute.json (hunk 3 above)
Comment (dagald) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/examples/createBasicAKSCompute.json]: Because if you look at our code (the MLC microservice code) and our internal swagger, we don't return a 200 anymore. We return 201 always for correct calls.
And the swagger needs to represent the rp correctly.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json (hunk 1 above)
Comment (dagald) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: they were not on our internal swagger. I can add the long-running back.

Delete is returning a 202

Comment (dagald) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: added long-running back. Update accordingly

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json (hunk 4 above)
Comment (dagald) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: added long-running back. Update accordingly

**Comment by dsgouda** [comment]
In case changes were made to the service itself we might need an ARM review here

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json [hunk 5]
```diff
@@ -608,15 +608,68 @@
             }
           }
         }
+      },
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: What's the rationale for adding a POST here? In any case, POST requires an action. This is invalid at `/workspaces/{workspaceName}/computes/{computeName}` without an action.

**Review by KrisBash — CHANGES_REQUESTED** [review]
See comments regarding that POST on /computes/{computeName}

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/examples/createBasicAKSCompute.json (hunk 3 above)
Comment (asjindal3312) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/examples/createBasicAKSCompute.json]: I see there are differences some files checked in to MLCompute repo and here. Can you make sure that both place we have the same swagger files?

Comment (dagald) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/examples/createBasicAKSCompute.json]: yes, after I get this checked in the azure repo, I will backport it to our repo.
But, for the future, I think that we should not maintain our own swagger, and everything should be changed here.
There is not sense in maintaining a swagger in both places

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json (hunk 5 above)
Comment (asjindal3312) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: It seems that the REST methods here are not consistent with REST API protocol and methods. How are these methods decided at the first place?

**Review by asjindal3312 — APPROVED** [review]

**Review by KrisBash — APPROVED** [review]
Approving from ARM on the understanding that this swagger update is aligning to actual service implementation. Comments can be addressed in a future API version.

**Review by dsgouda — APPROVED** [review]
LGTM
